distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso

```diff
@@ -193,8 +194,55 @@ type Date
          other hand, the first day of the week is Monday, and week 1 is the week
          containing the first Thursday of the year. Therefore it is important to
          properly specify the `locale` argument.
-    week_of_year : Locale.Locale -> Integer
-    week_of_year self locale=Locale.default = Time_Utils.week_of_year_localdate self locale.java_locale
+    week_of_year : (Locale.Locale | Nothing) -> Integer
+    week_of_year self locale=Nothing =
+        if locale.is_nothing then
+            Time_Utils.week_of_year_localdate self
+        else
+            Time_Utils.week_of_year_localdate self locale.java_locale
+
+
+    ## Returns if the date is in a leap year.
+    is_leap_year : Boolean
+    is_leap_year self =
+        year = self.year
+        (year%4 == 0) && ((year%100 != 0) || (year%400 == 0))
```

Comment:
This will not work for years prior to the transition to the gregorian calendar.

Why not use [`GregorianCalendar.isLeapYear`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/GregorianCalendar.html#isLeapYear(int)) from Java's stdlib?